source/common/router/scoped_rds.cc

```diff
@@ -173,20 +175,23 @@ bool ScopedRdsConfigSubscription::addOrUpdateScopes(
       route_provider_by_scope_.insert({scope_name, std::move(rds_config_provider_helper)});
       scope_name_by_hash_[scoped_route_info->scopeKey().hash()] = scoped_route_info->scopeName();
       scoped_route_map_[scoped_route_info->scopeName()] = scoped_route_info;
-      applyConfigUpdate([scoped_route_info](ConfigProvider::ConfigConstSharedPtr config)
-                            -> ConfigProvider::ConfigConstSharedPtr {
-        auto* thread_local_scoped_config =
-            const_cast<ScopedConfigImpl*>(static_cast<const ScopedConfigImpl*>(config.get()));
-        thread_local_scoped_config->addOrUpdateRoutingScope(scoped_route_info);
-        return config;
-      });
+      updated_scopes.push_back(scoped_route_info);
       any_applied = true;
       ENVOY_LOG(debug, "srds: add/update scoped_route '{}', version: {}",
```

Comment:
I suggest slightly changing the wording of this log message to clarify that the config has not yet been applied. e.g., "srds: queueing add/update of scoped_route ..."

Reply:
This makes sense, the changes are applied after the log.